rspace/src/main/scala/coop/rchain/rspace/LockFreeInMemoryStore.scala

```diff
@@ -0,0 +1,253 @@
+package coop.rchain.rspace
+
+import coop.rchain.rspace.history.{Branch, ITrieStore}
+import coop.rchain.rspace.internal._
+import coop.rchain.rspace.util.canonicalize
+import coop.rchain.shared.SeqOps.{dropIndex, removeFirst}
+import kamon._
+import scodec.Codec
+
+import scala.collection.concurrent.TrieMap
+import scala.collection.immutable.Seq
+
+class NoopTxn[S] extends InMemTransaction[S] {
+  override def commit(): Unit                   = ()
+  override def abort(): Unit                    = ()
+  override def close(): Unit                    = ()
+  override def readState[R](f: S => R): R       = ???
```

Comment:
I understand that `???` will be removed when `InMemTransaction` can be refactored after the removal of the old implementation